### PR TITLE
add custom width support for dialogs

### LINKED ISSUE #10630: ModelView wizards are extremely wide
The Configure Python dialog currently opens as 1200px wide, but the contents of the wizard pages are only 500-700px wide. The buttons at the bottom of the wizard are also only around 300px in width.

### PR DESCRIPTION
**Pull request by alanrenmsft**
This PR fixes #10630 

`
azdata.window.createWizard(title, 1800);
`
Screen Shot 2020-05-29 at 8 03 13 PM

`
azdata.window.createWizard(title, ''narrow');
`
Screen Shot 2020-05-29 at 8 00 54 PM

`
azdata.window.createWizard(title, 'medium');
`
Screen Shot 2020-05-29 at 8 01 42 PM

`
azdata.window.createWizard(title, 'wide');
`
Screen Shot 2020-05-29 at 8 02 23 PM

### CODE REVIEW

File: src/sql/workbench/browser/modal/modal.ts [hunk 1]
```diff
@@ -175,8 +177,8 @@ export abstract class Modal extends Disposable implements IThemable {
 			builderClass += ' flyout-dialog';
 		}
 
-		if (this._modalOptions.isWide) {
-			builderClass += ' wide';
+		if (this._modalOptions.width && typeof this._modalOptions.width === 'string') {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modal/modal.ts:180]: Probably easier to just  set the width here for all options and just switch on the different possible values instead of using css rules for some and  specifically setting the width for others

File: src/sql/workbench/api/browser/mainThreadModelViewDialog.ts [hunk 2]
```diff
@@ -213,7 +214,7 @@ export class MainThreadModelViewDialog implements MainThreadModelViewDialogShape
 
 	public $openWizard(handle: number): Thenable<void> {
 		let wizard = this.getWizard(handle);
-		this._dialogService.showWizard(wizard);
+		this._dialogService.showWizard(wizard, { hasBackButton: false, width: wizard.width, hasErrors: true });
```
Comment (Charles-Gagnon) [inline src/sql/workbench/api/browser/mainThreadModelViewDialog.ts:217]: Where these the defaults before for these other options?

File: src/sql/workbench/api/common/extHostModelViewDialog.ts [hunk 3]
```diff
@@ -688,13 +714,13 @@ export class ExtHostModelViewDialog implements ExtHostModelViewDialogShape {
...
 
-	public createDialog(title: string, dialogName?: string, extension?: IExtensionDescription, isWide?: boolean): azdata.window.Dialog {
+	public createDialog(title: string, dialogName?: string, extension?: IExtensionDescription, width?: azdata.window.DialogWidth): azdata.window.Dialog {
 		let dialog = new DialogImpl(this, this._extHostModelView, this._extHostTaskManagement, extension);
 		if (dialogName) {
 			dialog.dialogName = dialogName;
 		}
 		dialog.title = title;
-		dialog.isWide = isWide;
+		dialog.width = width === undefined ? 'narrow' : width;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/api/common/extHostModelViewDialog.ts:723]: FYI you can shorten this to `dialog.width = width ?? 'narrow'` since that'll still allow falsy values like 0.

File: src/sql/workbench/api/common/extHostModelViewDialog.ts [hunk 4]
```diff
@@ -653,9 +671,17 @@ export class ExtHostModelViewDialog implements ExtHostModelViewDialogShape {
 		}
 		this.updateButton(dialog.okButton);
 		this.updateButton(dialog.cancelButton);
+		let dialogWidth: DialogWidth = 'narrow';
```
Comment (Charles-Gagnon) [inline src/sql/workbench/api/common/extHostModelViewDialog.ts:674]: Here you have the default be narrow but everywhere else it seems to be wide - why the difference?

File: src/sql/workbench/api/common/sqlExtHost.api.impl.ts [hunk 5]
```diff
@@ -402,8 +402,14 @@ export function createAdsApiFactory(accessor: ServicesAccessor): IAdsExtensionAp
 				createWebViewDialog(name: string) {
 					return extHostModalDialogs.createDialog(name);
 				},
-				createModelViewDialog(title: string, dialogName?: string, isWide?: boolean): azdata.window.Dialog {
-					return extHostModelViewDialog.createDialog(title, dialogName, extension, !!isWide);
+				createModelViewDialog(title: string, dialogName?: string, width?: boolean | azdata.window.DialogWidth): azdata.window.Dialog {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/api/common/sqlExtHost.api.impl.ts:405]: Maybe put a comment here explaining that the boolean is for backwards compat support

File: src/sql/azdata.proposed.d.ts [hunk 6]
```diff
@@ -360,6 +360,37 @@ declare module 'azdata' {
...
+
+		export type DialogWidth = 'narrow' | 'medium' | 'wide' | number;
+
+		/**
+		 * Create a dialog with the given title
+		 * @param title The title of the dialog, displayed at the top
+		 * @param dialogName the name of the dialog
+		 * @param width width of the dialog, default is 'wide'
+		 */
+		export function createModelViewDialog(title: string, dialogName?: string, width?: DialogWidth): Dialog;
```
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:386]: We should figure out how we want to handle deprecating stuff - but I feel like a @deprecated annotation on the azdata.d.ts one would be a good idea.

File: src/sql/workbench/api/browser/mainThreadModelViewDialog.ts (hunk 2 above)
Comment (alanrenmsft) [reply src/sql/workbench/api/browser/mainThreadModelViewDialog.ts:217]: i have updated the code to reflect the source.

File: src/sql/workbench/api/common/extHostModelViewDialog.ts (hunk 4 above)
Comment (alanrenmsft) [reply src/sql/workbench/api/common/extHostModelViewDialog.ts:674]: not really, Dialog and Wizard has different default values, for dialogs, we used to allow authors to specify isWide option, and the default is false, the new default value is 'narrow', for wizards, we only support wide dialog, so the new default value is 'wide'.
